Interact with search web page (WebView)

woojn90 opened the pull request:
- [x] #12 Add search keyword (UI, DB) & search engine
- [x] #7 Connect to web search & Get data from url (save to app)

woojn90 committed:
Update README.md

woojn90 committed:
commit

add favorite star image
apply search word (UI, DB)
add docs (just copy)

woojn90 committed:
solve the conflict problem

README.md

dalinaum commented on app/src/main/java/com/android/woojn/coursebookmarkapplication/CourseActivity.java:
```diff
@@ -101,27 +111,27 @@ public void onItemClick(int id, int viewId) {
                 break;
             case R.id.btn_delete_section:
-                Log.d("Check", "delete / section id : " + id);
                 // TODO: alert 띄우고 삭제하도록 수정
```
Alert보다는 다이얼로그가 낫겠지요?

woojn90 replied:
네 AlertDialog 사용하도록 하겠습니다~

dalinaum commented on docs/LICENSE.txt:
```diff
@@ -0,0 +1,63 @@
+Creative Commons Attribution 3.0 Unported
```
음 CC라이선스가 여기에 필요하나요?

woojn90 replied:
docs 는 지난주 특강때 윤성관 튜터님께 들은 내용으로, 앱 홍보 url을 만드는 방법입니다!
master에 docs가 생기면 해당 github에서 제공하는 url로 접근이 가능하다고 하네요ㅎㅎ
그래서 미리 추가만 해봤습니다. (마지막 주에 수정할 예정)

dalinaum reviewed (approved):
docs를 제외한 부분에 대해서는 LGTM 👍이고 Approved합니다. docs는 어떤 역활인가요? (이건 커멘트만 달아주세요.)

woojn90 commented:
아직 in progress라서 review로 바꾼 후 다시 리뷰 요청드리겠습니다.

woojn90 committed:
Commit

[As-is]
Lint 완료 (90%)
Button Image 추가 및 style, dimen 적용
UI 틀 변경 및 발견한 bug 잡음
[To-do]
Overlay button 적용 및 URL preview 받기

woojn90 committed:
commit

- TabHost -> TabLayout & ViewPager
- 검색엔진 (instagram) 추가

dalinaum commented on app/src/main/java/com/android/woojn/coursebookmarkapplication/adapter/PagerAdapter.java:
```diff
@@ -0,0 +1,42 @@
+
+    public PagerAdapter(FragmentManager fm, int tabCount) {
+        super(fm);
+        this.mTabCount = tabCount;
```
`this` 제거

dalinaum commented on app/src/main/java/com/android/woojn/coursebookmarkapplication/fragment/CourseFragment.java:
```diff
@@ -0,0 +1,131 @@
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.fragment_course, null);
```
두버뻔쨰 인자로 `container`를 전달해봅시다.

dalinaum commented on app/src/main/java/com/android/woojn/coursebookmarkapplication/fragment/CourseFragment.java:
```diff
@@ -0,0 +1,131 @@
+        mRealm.executeTransaction(new Realm.Transaction() {
+            @Override
+            public void execute(Realm realm) {
+                Course course = realm.where(Course.class).equalTo("id", courseId).findFirst();
```
해당 부분을 트랜잭션 밖으로 옮기자.

dalinaum commented on app/src/main/java/com/android/woojn/coursebookmarkapplication/fragment/CourseFragment.java:
```diff
@@ -0,0 +1,131 @@
+                mRealm.executeTransaction(new Realm.Transaction() {
+                    @Override
+                    public void execute(Realm realm) {
+                        Course course = realm.where(Course.class).equalTo("id", courseId).findFirst();
```
뺄 수 있는지 알아보자.

woojn90 committed:
final commit

[주요]
- Floating Button service 제거 -> WebView로 Url 저장 완료
- html 파싱 완료(80% / 나머지는 #20 으로 보류)
- 이미지 로딩 (Glide) 완료

[기타]
- ConstantClass 추가
- 불필요 this 제거 등 튜터님 review 보완

dalinaum commented on app/src/main/java/com/android/woojn/coursebookmarkapplication/ConstantClass.java:
```diff
@@ -0,0 +1,17 @@
+ * Created by wjn on 2017-02-17.
+ */
+
+public class ConstantClass {
```
Constants

dalinaum commented on app/src/main/java/com/android/woojn/coursebookmarkapplication/ConstantClass.java:
```diff
@@ -0,0 +1,17 @@
+
+public class ConstantClass {
+    public static final String ID = "id";
+    public static final String COURSE_ID = "courseId";
```
`KEY_`

dalinaum commented on app/src/main/java/com/android/woojn/coursebookmarkapplication/ConstantClass.java:
```diff
@@ -0,0 +1,17 @@
+public class ConstantClass {
+    public static final String ID = "id";
+    public static final String COURSE_ID = "courseId";
+    public static final String SECTION_ID = "sectionId";
```
Same above.

dalinaum commented on app/src/main/java/com/android/woojn/coursebookmarkapplication/ConstantClass.java:
```diff
@@ -0,0 +1,17 @@
+    public static final String ID = "id";
+    public static final String COURSE_ID = "courseId";
+    public static final String SECTION_ID = "sectionId";
+    public static final String STRING_URL = "stringUrl";
```
`KEY_`

dalinaum commented on app/src/main/java/com/android/woojn/coursebookmarkapplication/ConstantClass.java:
```diff
@@ -0,0 +1,17 @@
+    public static final String COURSE_ID = "courseId";
+    public static final String SECTION_ID = "sectionId";
+    public static final String STRING_URL = "stringUrl";
+    public static final String REQUEST_WEB_ACTIVITY = "requestWebActivity";
```
Same above

dalinaum commented on app/src/main/java/com/android/woojn/coursebookmarkapplication/ConstantClass.java:
```diff
@@ -0,0 +1,17 @@
+    public static final String STRING_URL = "stringUrl";
+    public static final String REQUEST_WEB_ACTIVITY = "requestWebActivity";
+
+    public static final int VIEW_ID_OF_ITEM_VIEW = 0;
```
`DEFAULT_`

dalinaum commented on app/src/main/java/com/android/woojn/coursebookmarkapplication/ConstantClass.java:
```diff
@@ -0,0 +1,17 @@
+ */
+
+public class ConstantClass {
+    public static final String ID = "id";
```
`FIELD_NAME_`

dalinaum commented on app/src/main/java/com/android/woojn/coursebookmarkapplication/ConstantClass.java:
```diff
@@ -0,0 +1,17 @@
+    public static final String REQUEST_WEB_ACTIVITY = "requestWebActivity";
+
+    public static final int VIEW_ID_OF_ITEM_VIEW = 0;
+    public static final int REQUEST_WEB_ACTIVITY_WITH_SAVE = 0;
```
임의의 값으로 설정.

dalinaum commented on app/src/main/java/com/android/woojn/coursebookmarkapplication/ConstantClass.java:
```diff
@@ -0,0 +1,17 @@
+
+    public static final int VIEW_ID_OF_ITEM_VIEW = 0;
+    public static final int REQUEST_WEB_ACTIVITY_WITH_SAVE = 0;
+    public static final int REQUEST_WEB_ACTIVITY_WITHOUT_SAVE = 1;
```
REQUEST_WEB_ACTIVITY_WITHOUT_SAVE = REQUEST_WEB_ACTIVITY_WITH_SAVE + 1

dalinaum commented on app/src/main/java/com/android/woojn/coursebookmarkapplication/activity/CourseActivity.java:
```diff
@@ -420,4 +446,54 @@ public void afterTextChanged(Editable s) {
+                }
+
+                realm.beginTransaction();
+                for (int i = 0; i < ogTags.size(); i++) {
```
for each가 안되나요?

dalinaum commented on app/src/main/java/com/android/woojn/coursebookmarkapplication/activity/CourseActivity.java:
```diff
@@ -420,4 +446,54 @@ public void afterTextChanged(Editable s) {
         editTextSearchWord.addTextChangedListener(textWatcher);
     }
 
+    private void setSectionDetailByUrl(int sectionDetailId) {
```
retrieveDetailSectionById

dalinaum commented on app/src/main/java/com/android/woojn/coursebookmarkapplication/activity/CourseActivity.java:
```diff
@@ -420,4 +446,54 @@ public void afterTextChanged(Editable s) {
 
+    private void setSectionDetailByUrl(int sectionDetailId) {
+        // TODO: refresh (이미 로딩된 항목 처리)
+        JsoupAsyncTask jsoupAsyncTask = new JsoupAsyncTask();
```
ParseAsyncTask

dalinaum commented on app/src/main/java/com/android/woojn/coursebookmarkapplication/activity/CourseActivity.java:
```diff
@@ -93,17 +108,26 @@ protected void onCreate(Bundle savedInstanceState) {
+        for (Section section : mCourse.getSections()) {
+            for (SectionDetail sectionDetail : section.getSectionDetails()) {
+                int sectionDetailId = sectionDetail.getId();
+                setSectionDetailByUrl(sectionDetailId);
```
`visted` 필드를 넣어서 방문했던 곳을 일단 배제를 시켜보죠.

dalinaum commented on app/src/main/java/com/android/woojn/coursebookmarkapplication/activity/CourseActivity.java:
```diff
@@ -288,6 +289,24 @@ public void execute(Realm realm) {
         builder.show();
     }
 
+    private void searchByWebView(int sectionId) {
```
searchAndShowResults

dalinaum commented on app/src/main/java/com/android/woojn/coursebookmarkapplication/activity/WebActivity.java:
```diff
@@ -0,0 +1,151 @@
+        }
+
+        mSectionId = getIntent().getIntExtra(SECTION_ID, 0);
+        String stringUrl = getIntent().getStringExtra(STRING_URL);
```
가져온 번들 값 체크가 필요.

dalinaum commented on app/src/main/java/com/android/woojn/coursebookmarkapplication/activity/WebActivity.java:
```diff
@@ -0,0 +1,151 @@
+        mWebView.getSettings().setJavaScriptEnabled(true);
+        mWebView.loadUrl(stringUrl);
+        mWebView.setWebViewClient(new WebViewClient() {
+            @SuppressWarnings("deprecation")
```
다른 방법이 있는지 확인해보시죠.

dalinaum replied:
Nvm

dalinaum commented on app/src/main/java/com/android/woojn/coursebookmarkapplication/activity/WebActivity.java:
```diff
@@ -0,0 +1,151 @@
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                NavUtils.navigateUpFromSameTask(this);
```
뺼 수 있을 듯.

dalinaum commented on app/src/main/java/com/android/woojn/coursebookmarkapplication/activity/WebActivity.java:
```diff
@@ -0,0 +1,151 @@
+        super.onBackPressed();
+    }
+
+    private void makeToastAfterCancel(int resId) {
```
showToastByForce

dalinaum commented on app/src/main/java/com/android/woojn/coursebookmarkapplication/adapter/CourseSectionAdapter.java:
```diff
@@ -84,27 +90,22 @@ public CourseSectionViewHolder(View itemView) {
-                    SectionDetail sectionDetail = realm.where(SectionDetail.class).equalTo("id", id).findFirst();
-                    String url = sectionDetail.getUrl();
+                    SectionDetail sectionDetail = realm.where(SectionDetail.class).equalTo(ID, id).findFirst();
                     realm.close();
```
realm.close()는 `sectionDetail.getUrl();`

dalinaum commented on app/src/main/java/com/android/woojn/coursebookmarkapplication/fragment/CourseFragment.java:
```diff
@@ -93,11 +103,10 @@ private void makeToastAfterCancel(int resId) {
     }
 
-    private void updateCourseFavoriteById(final int courseId) {
+    private void updateCourseFavoriteById(final Course course) {
```
toggleCourseFavorited

woojn90 committed:
apply tutor's review

dalinaum reviewed (approved):
LGTM 👍
